packages/apputils/src/sessioncontext.tsx

```diff
@@ -337,6 +343,11 @@ export class SessionContext implements ISessionContext {
    * Construct a new session context.
    */
   constructor(options: SessionContext.IOptions) {
+    this.kernelManager =
+      options.kernelManager ??
+      new KernelManager({
```

Comment:
We should remove this and make `options.kernelManager` required for the next major version of `@jupyterlab/apputils`